### PR TITLE
starter code for c++

### PR DESCRIPTION
**Pull request by TilakChad**
## Checklist 

- [x] My code follows the style guidelines of this project
- [#] I have performed a self-review of my own code
- [#] I have commented my code, particularly in hard-to-understand areas
- [#] I have made corresponding changes to the documentation (if necessary)
- [#] I have performed adequate tests that prove my fix is effective or that my feature works

## Proposed Changes
This pull request proposes to include provided code as c++ starter code for bhoos smart bot 2023 competition.

### CODE REVIEW

File: cpp/src/bot.cpp [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    return Suit(mycards[3].suit);
+}
+
+int GameState::Bid(PlayerID myid, std::vector<PlayerID> player_ids, std::vector<Card> mycards, int32_t time_remaining,
+                   std::vector<BidEntry> bid_history, BidState const &bid_state)
+{
+    // Either bid or pass
+    if (bid_state.defender.bid_value == -1 && bid_state.challenger.bid_value == -1)
+        return 16;
+    return 0;
```
Comment (pSapien) [inline cpp/src/bot.cpp:85]: @TilakChad Please refrain from using  `-1` as an indicator to pass the bid. 
Check whether `bidHistory` is empty or not to pass the minimum bid.

`-1` may change.

File: cpp/Readme.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# Starter Code for Bhoos's Smart Bots 2023 
+
+Still wondering why the language that runs the world is missing from this grand competition? <br>
+Now's probably the right time to stop wondering, (>.<) <br>
```
Comment (pSapien) [inline cpp/Readme.md:4]: @TilakChad We were waiting for chad to show up.

File: cpp/src/bot.cpp (hunk 1 above)
Comment (TilakChad) [reply cpp/src/bot.cpp:85]: I've  modified that to check for empty bid history now. I was following the API docs closely, so thought that would be  the way. Other things like the order of json element weren't were exactly like as the API docs enlisted. So had to consider that too.

File: cpp/Readme.md (hunk 2 above)
Comment (TilakChad) [reply cpp/Readme.md:4]: xD

File: cpp/src/bot.cpp (hunk 1 above)
Comment (pSapien) [reply cpp/src/bot.cpp:85]: @TilakChad Other things like the order of json element weren't were exactly like as the API docs enlisted. So had to consider that too.

Could you elaborate a bit more on this?

Comment (TilakChad) [reply cpp/src/bot.cpp:85]: > @TilakChad Other things like the order of json element weren't were exactly like as the API docs enlisted. So had to consider that too.
> 
> Could you elaborate a bit more on this?

I was referring to this part :
image

and the actual payload  : 
image

As far as it concerns API documentation, this is an oversight. 

Faced onto this while parsing json sequentially on that order, which would make parsing a bit faster. Usual way is indexing onto the property id : value, which isn't terrible but okish.

**Comment by pSapien** [comment]
Thanks for the PR

**Pull request merged**